clang/test/CodeGenOpenCL/builtins-amdgcn.cl

```diff
@@ -896,5 +896,6 @@ void test_set_fpenv(unsigned long env) {
   __builtin_amdgcn_set_fpenv(env);
 }
 
+// CHECK-DAG: [[$GRID_RANGE]] = !{i32 1, i32 0}
```

Comment:
the upper bound is smaller than the lower bound?

Reply:
Yes, this is how you are supposed to represent the wrapped set where the 0 value isn't allowed but the uint32_max is

Reply:
oh I c. range does allow wrap.